File: java/Vulnerable.java
/**
 * Vulnerable Java code samples for SAST testing
 * Author: Test Suite
 * Created: 2025-01-21
 */

import java.sql.*;
import java.io.*;
import java.nio.file.*;
import java.security.MessageDigest;

public class Vulnerable {
    
    // SQL Injection vulnerability
    public ResultSet getUserData(String userId) throws SQLException {
        String query = "SELECT * FROM users WHERE id = " + userId;
        Statement stmt = connection.createStatement();
        return stmt.executeQuery(query);
    }
    
    // SQL Injection with string formatting
    public ResultSet searchUsers(String searchTerm) throws SQLException {
        String query = String.format("SELECT * FROM users WHERE name LIKE '%s'", searchTerm);
        Statement stmt = connection.createStatement();
        return stmt.executeQuery(query);
    }
    
    // Command Injection
    public void executeCommand(String command) throws IOException {
        Runtime.getRuntime().exec(command);
    }
    
    // Hardcoded secrets
    private static final String API_KEY = "sk_live_51H3ll0W0rld123456789";
    private static final String PASSWORD = "admin123";
    private static final String SECRET = "mySecretKey123";
    
    // Path Traversal
    public String readFile(String filename) throws IOException {
        return new String(Files.readAllBytes(Paths.get("/var/www/data/" + filename)));
    }
    
    // Weak crypto - MD5
    public String hashPassword(String password) throws Exception {
        MessageDigest md = MessageDigest.getInstance("MD5");
        byte[] hash = md.digest(password.getBytes());
        return bytesToHex(hash);
    }
    
    // Insecure deserialization
    public Object deserializeData(byte[] data) throws IOException, ClassNotFoundException {
        ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(data));
        return ois.readObject();
    }
    
    // XSS vulnerability
    public void displayMessage(String message) {
        response.getWriter().println("<div>" + message + "</div>");
    }
    
    private String bytesToHex(byte[] bytes) {
        StringBuilder sb = new StringBuilder();
        for (byte b : bytes) {
            sb.append(String.format("%02x", b));
        }
        return sb.toString();
    }
}

File: kotlin/Vulnerable.kt
/**
 * Vulnerable Kotlin code samples for SAST testing
 * Author: Test Suite
 * Created: 2025-01-21
 */

import java.security.MessageDigest
import java.io.File
import java.sql.Statement

// SQL Injection vulnerability
fun getUserData(userId: String): ResultSet {
    val query = "SELECT * FROM users WHERE id = $userId"
    val stmt: Statement = connection.createStatement()
    return stmt.executeQuery(query)
}

// SQL Injection with string interpolation
fun searchUsers(searchTerm: String): ResultSet {
    val query = "SELECT * FROM users WHERE name LIKE '$searchTerm'"
    val stmt: Statement = connection.createStatement()
    return stmt.executeQuery(query)
}

// Command Injection
fun executeCommand(command: String) {
    Runtime.getRuntime().exec(command)
}

// Hardcoded secrets
const val API_KEY = "sk_live_51H3ll0W0rld123456789"
const val PASSWORD = "admin123"
const val SECRET = "mySecretKey123"

// Path Traversal
fun readFile(filename: String): String {
    return File("/var/www/data/$filename").readText()
}

// Weak crypto - MD5
fun hashPassword(password: String): String {
    val md = MessageDigest.getInstance("MD5")
    val hash = md.digest(password.toByteArray())
    return hash.joinToString("") { "%02x".format(it) }
}

// Insecure deserialization
fun deserializeData(data: ByteArray): Any {
    val ois = java.io.ObjectInputStream(java.io.ByteArrayInputStream(data))
    return ois.readObject()
}

// XSS vulnerability
fun displayMessage(message: String) {
    response.writer.println("<div>$message</div>")
}

